catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/sharing/ShareableMetacardImpl.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.sharing;
+
+import com.google.common.collect.Sets;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import ddf.catalog.data.impl.types.CoreAttributes;
+import ddf.catalog.data.impl.types.SecurityAttributes;
+import ddf.catalog.data.types.Core;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/** A metacard that supports sharing. */
+public class ShareableMetacardImpl extends MetacardImpl {
+
+  public ShareableMetacardImpl(MetacardType type) {
+    super(type);
+  }
+
+  public ShareableMetacardImpl(Metacard metacard) {
+    super(metacard);
+  }
+
+  public ShareableMetacardImpl() {
+    super(
+        new MetacardTypeImpl(
+            "metacard.sharing", Arrays.asList(new CoreAttributes(), new SecurityAttributes())));
+  }
+
+  public ShareableMetacardImpl(String id) {
+    this();
+    setId(id);
+  }
+
+  /**
+   * Check if a given metacard is a shareable metacard by checking the tags metacard attribute.
```

Comment:
❓ If we're not defining any new attributes, does this tag restriction still make sense?
✏️ I'd prefer to reference attributes (`metacard-tags`) either in their API declaration via javadoc, or at least as their literal value.